spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/sql/TestPartitionedWrites.java

```diff
@@ -31,11 +34,35 @@
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runners.Parameterized;
 
 public class TestPartitionedWrites extends SparkCatalogTestBase {
+
+  private String targetBranch;
+
   public TestPartitionedWrites(
-      String catalogName, String implementation, Map<String, String> config) {
+      String catalogName, String implementation, Map<String, String> config, String targetBranch) {
```

Comment:
I'd prefer to copy the tests and add the VERSION AS OF to the copies rather than adding a test param. Basically using the same logic as my comment above, we still want to exercise the path that _doesn't_ use VERSION AS OF.